### DISCUSSION TITLE
Switching to _tilt Features

### DISCUSSION THREAD

**Discussion by r01k** [r01k/ha_sunsa#2] [Ideas]
Discussion originally started in the [Home Assistant Community](https://community.home-assistant.io/t/sunsa-blinds-wands-integration/708429/8?u=odwide).

Currently ([v1.0.0](https://github.com/r01k/ha_sunsa/releases/tag/v1.0.0)) the integration supports "non-`_tilt`" features:

https://github.com/r01k/ha_sunsa/blob/c42c8a269dcde7dfddd7c090536e12f385ff1b5c/custom_components/sunsa/cover.py#L75-L79

`cover` entities can also support "`_tilt`" [features](https://www.home-assistant.io/integrations/cover/#services), which are functionally similar but result in a slightly different representation in the frontend. The following would apply to the Sunsa integration:
- Services:
  - `cover.open_cover_tilt`
  - `cover.close_cover_tilt`
  - `cover.toggle_tilt`
  - `cover.set_cover_tilt_position`
- Attributes:
  - `current_tilt_position`

They would provide a nice visual gradient of the cover positioning in the frontend:

![image](https://github.com/r01k/ha_sunsa/assets/19595786/a454d38b-34ff-4788-84c3-99bf4ceee7ff) | ![image](https://github.com/r01k/ha_sunsa/assets/19595786/324a5187-23b3-4bd5-b36b-189080d91704)

![image](https://github.com/r01k/ha_sunsa/assets/19595786/b11a44e1-c587-49d1-98d0-b0528921f6c7)

But switching to `_tilt` would break existing installations until users manually change the service and attribute calls. Unless `_tilt` features are instead added with the current non-`_tilt` ones, which would allow to use either. The problem then is that it'll be confusing to tell what's the difference (there's no functional difference) and, even worse, the `More info` screen will show both sliders:

![image](https://github.com/r01k/ha_sunsa/assets/19595786/e05d760b-fb1c-424d-be6c-22b75633571f)

A breaking change notice could be added in the release notes a few months in advance of removing the non-`_tilt` features.

I really like the visual gradient that `_tilt` provides and think the change is worth it. If someone else agrees, let me know.

**Comment by N3rd7** [comment]
Hi, thank you for developing this integration, it works amazingly! I've become increasingly interested in having the blinds tilt progressively with the sun. There is an integration for that, but I think it requires tilt all around (i.e.: have the app control and report closed all the way up, open (horizontal slats) and closed all the way down. Or maybe just tilt from closed to open 100% in one direction only, I'm not sure. 
Anyway, is there a way I can do that in home assistant by chance?

**Reply by r01k** [reply]
Hey there. You can use the action `sunsa.set_absolute_position` to set the blinds (horizontal or vertical) to any position (-100 - 100). But not sure if that integration would support calling discrete actions.
